src/Migrations/M053CustomisePostsListSeeAllLink.php

```diff
@@ -0,0 +1,152 @@
+            return $innerBlock['blockName'] === Utils\Constants::BLOCK_NAV_LINK &&
+                $innerBlock['attrs']['className'] === 'see-all-link';
+        });
+        if (!$see_all_link) {
+            return false;
+        }
+
+        // Check if the block has taxonomy filters.
+        return isset($block['attrs']['query']['taxQuery']);
+    }
+
+    /**
+     * Update "see all" link to add the taxonomy filters.
+     *
+     * @param array $block - A block data array.
+     * @return array - The transformed block.
+     */
+    private static function transform_block(array $block): array
+    {
+        // Get the News & Stories page url.
+        $news_page_url = get_permalink(self::news_page_id());
+
+        // Create the new "see all" URL based on taxonomy filters.
+        $new_see_all_url = $news_page_url . '?';
+        $filters = $block['attrs']['query']['taxQuery'];
+        $tag_id = isset($filters['post_tag']) ? (int) $filters['post_tag'][0] : null;
+        $category_id = isset($filters['category']) ? (int) $filters['category'][0] : null;
+        $post_type_id = isset($filters['p4-page-type']) ? (int) $filters['p4-page-type'][0] : null;
+
+        // Add tag filter
+        if ($tag_id) {
+            $tag = get_term_by('id', $tag_id, 'post_tag');
+            $new_see_all_url .= 'tag=' . $tag->slug . '&';
+        }
+        // Add category filter
+        if ($category_id) {
+            $category = get_term_by('id', $category_id, 'category');
+            $new_see_all_url .= 'category=' . $category->slug . '&';
+        }
+        // Add post type filter
+        if ($post_type_id) {
+            $post_type = get_term_by('id', $post_type_id, 'p4-page-type');
+            $new_see_all_url .= 'post-type=' . $post_type->slug;
+        }
+
+        // Update the "see all" link in all navigation links.
+        // We remove the ending "&" or "?" character if needed.
+        self::update_see_all_links($block['innerBlocks'], rtrim($new_see_all_url, '&?'));
+
+        return $block;
+    }
+
+    /**
+     * Returns the News & Stories page id from the settings.
+     */
+    private static function news_page_id(): int
+    {
+        if (self::$news_page_id === null) {
+            self::$news_page_id = (int) get_option('page_for_posts');
+        }
+        return self::$news_page_id;
+    }
+
+    /**
+     * Update the "see all" url in all navigation link blocks, by default the Posts List block has two.
+     *
+     * @param array $blocks - The blocks to go through and potentially update.
+     * @param string $new_link - The new url value.
+     */
+    private static function update_see_all_links(array &$blocks, string $new_link): void
+    {
+        foreach ($blocks as &$block) {
+            if (
+                $block['blockName'] === Utils\Constants::BLOCK_NAV_LINK &&
+                $block['attrs']['className'] === 'see-all-link'
+            ) {
+                $block['attrs']['url'] = $new_link;
```

Comment:
So I have a question here, because the last time this was brought up, I was DoD and suggested editors could edit the _See All_ link manually and a URL to a custom page, assuming some editors have done that, do we need to check first? Ccing @comzeradd for clarity too.

Reply:
Yes, but unfortunately we can't fit all use cases in the migration. So the ones who have edited the link to something completely different, we will send them a heads up to let me know that their changes will be overridden. See [this discussion](https://greenpeace-gpi.slack.com/archives/G015K63081W/p1751623439142169)